### ISSUE TITLE
Docker start fail on OSX Mojave

### ISSUE THREAD

**Issue by klyff** [docker/for-mac#3550]
- [ ] I have tried with the latest version of my channel (Stable or Edge)
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Information

  - macOS Version:

### Diagnostic logs

```
Docker for Mac: version...

```

### Steps to reproduce the behavior

  1. ...
  2. ...

**Comment by jcharnley** [comment]
Hey does your mac crash when you start up the application? I am getting a kernel panic error but I believe the problem is my side...

**Comment by kebeda** [comment]
I have this exact same problem
Diagnostic ID: C23FC371-7180-4898-A542-602826359632/20190227174214

**Comment by marcelomelodf** [comment]
I've been getting the same problem, docker desktop would not start (always with the error: `Supervisor has failed, shutting down: Supervisor caught an error: one of the children died: com.docker.vpnkit`) and diagnostics would  hang when I clicked on the 'Diagnose & Feedback' button. 
Reading the troubleshoot I ran diagnostics through the command line (ID: 118DEE9B-F4F1-400B-8356-438D5179D236/20190227191352) and then I tried to run `/Applications/Docker.app/Contents/MacOS/Docker --version` to see if I could at least double check the installed docker desktop version, and what was my surprise when docker started and ran without crashing!
I quit docker and tried running from Finder again, same thing, crashed with the same error. Quit and launched through CLI again: no crashes...
I am using Docker Desktop Version 2.0.0.3 (31259) and MacOS version 10.14.3

TL; DR; try starting docker through command line: `/Applications/Docker.app/Contents/MacOS/Docker`

**Comment by kebeda** [comment]
@marcelomelodf I tried this and it worked for me, thank you very much for the information. Hopefully Docker will fix this soon so we don't have to do this workaround...

**Comment by SooLee** [comment]
I'm having the same issue. `/Applications/Docker.app/Contents/MacOS/Docker` doesn't do anything.

**Comment by rvesse** [comment]
I had something like this and it was due to the underlying VM disk image that Docker uses getting corrupted.  I was able to fix it by going to `Preferences > Reset` and hitting the `Reset Disk Image` button which resolved my issues

**Comment by tbehling** [comment]
> I had something like this and it was due to the underlying VM disk image that Docker uses getting corrupted. I was able to fix it by going to `Preferences > Reset` and hitting the `Reset Disk Image` button which resolved my issues

Thank you for posting this @rvesse!  I had the symptom described above by @marcelomelodf, and not until I used `Reset Disk Image` did Docker start.

I believe this occurred after my Mac battery ran down to zero (I was racing the red battery meter, and lost!) the computer abruptly shut down, and I later plugged it in and restarted.  Maybe Docker didn't get a chance to gracefully unmount the VM's disk?  I wonder if Docker should have a check for this at boot to allow graceful recovery?

**Comment by shamill** [comment]
Found out it was a conflict with Visual Studio Code. 

"Docker is starting..." forever, never went away despite clearing all preferences, caches, and trying a different release.

Only when I closed Visual Studio Code was I then able to successfully start Docker.
